### PR TITLE
More detailed Fusion Reactor recipes NEI info

### PR DESCRIPTION
**Pull request by HoleFish**
More convenient for players to calculate overclock.

### CODE REVIEW

**Comment by Dream-Master** [comment]
@HoleFish guess the nei gui need to make a big bigger to hold all this info.

**Comment by miozune** [comment]
I think having two startup tier could be confusing. Doesn't recipe catalyst calculate overclock for you?

**Comment by HoleFish** [comment]
The old MK confused 'voltage tier' with 'capable tier'. Actually it has nothing to do with 'overclocking tier', which has confused a lot of players.(see discussion [here](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/13739))

**Comment by miozune** [comment]
I think I understand the situation now. Then how about appending `startupTier` to `Start:` property and using `startupTierForOverclock` to show "Higher tier machine than MkX can overclock" (or "This recipe cannot be overclocked") instead? I think that's more intuitive

File: src/main/java/gregtech/nei/formatter/FusionSpecialValueFormatter.java [hunk 1]
```diff
@@ -21,39 +21,66 @@ public class FusionSpecialValueFormatter implements INEISpecialInfoFormatter {
...
+                GT_Utility.formatNumbers(euToStart),
+                startupTierForOverclock));
+        specialInfo.add(StatCollector.translateToLocalFormatted("GT5U.nei.fusion_capable", startupTier));
+
+        return specialInfo;
+    }
+
+    public static int getFusionStartupTierForOverclock(long startupPower) {
+        // The same startup power data used in overclock calculation.
+        if (startupPower <= 160000000) {
```
Comment (miozune) [inline src/main/java/gregtech/nei/formatter/FusionSpecialValueFormatter.java:46]: Use underscore to make them more readable: `160_000_000`

**Review by miozune — COMMENTED** [review]
You can use `neiHandlerInfo` for recipemap to increase height of NEI display

**Comment by HoleFish** [comment]
Now NEI displays these (if there's no error in my code)

File: src/main/java/gregtech/api/objects/overclockdescriber/FusionOverclockDescriber.java [hunk 2]
```diff
@@ -7,17 +7,18 @@
...
 import gregtech.api.util.GT_Recipe;
-import gregtech.api.util.GT_Utility;
 import gregtech.api.util.MethodsReturnNonnullByDefault;
 
 @ParametersAreNonnullByDefault
 @MethodsReturnNonnullByDefault
 public class FusionOverclockDescriber extends EUOverclockDescriber {
 
     protected final long capableStartup;
+    protected final int tier;
```
Comment (miozune) [inline src/main/java/gregtech/api/objects/overclockdescriber/FusionOverclockDescriber.java:17]: This is the exact copy of `OverclockDescriber#tier`, you can use it instead

File: src/main/java/gregtech/api/objects/overclockdescriber/FusionOverclockDescriber.java [hunk 3]
```diff
@@ -43,10 +48,6 @@ public String getTierString() {
 
     @Override
     public boolean canHandle(GT_Recipe recipe) {
-        byte tier = GT_Utility.getTier(recipe.mEUt);
```
Comment (miozune) [inline src/main/java/gregtech/api/objects/overclockdescriber/FusionOverclockDescriber.java:46]: Any reason for removing this? e.g. Mk1 cannot handle titanium or oxygen plasma due to their voltage but they'll show up this way

Comment (HoleFish) [reply src/main/java/gregtech/api/objects/overclockdescriber/FusionOverclockDescriber.java:46]: An opposite example is that Compact Fusion MK4 can do Th/Pb/Pu-241 plasma, however they won't show up in MK4 map also due to their voltage. So I decide to show them all.
There may be a better way to deal with this conflict, but I don't quite have any idea.

**Comment by chochem** [comment]
This looks all wrong. There arent three different tiers for fusion recipes.

File: src/main/java/gregtech/api/objects/overclockdescriber/FusionOverclockDescriber.java (hunk 3 above)
Comment (miozune) [reply src/main/java/gregtech/api/objects/overclockdescriber/FusionOverclockDescriber.java:46]: You can set custom OverclockDescriber for compacts in GG code.
Oh also, GT++ sets its own describer `AdvancedFusionOverclockDescriber`, so `if(tier>8)` for `getEUtIncreasePerOC` and `getDurationDecreasePerOC` are not needed if you set them differently for `LargeFusionComputerPP`.

**Review by miozune — APPROVED** [review]

**Comment by miozune** [comment]
@chochem Did you read the code? There are actually 3 factors to account for when making plan for compact fusions

**Comment by chochem** [comment]
But the player makes normal fusion first, not compact. This new page is terrible and confusing for players getting to know fusion.

**Comment by HoleFish** [comment]
The old MK says Rhugnor recipe is MK4, however it can never be overclocked even in Compact MK5. That's what's even more confusing. At least NEI should tell players how it actually works instead of hiding all information about overclocking in the code.

**Comment by chochem** [comment]
It tells you the needed tier and is correct. Also the numbers to calculate OCS. Nothing is hidden in the code.

**Review by chochem — CHANGES_REQUESTED** [review]
These changes are incorrect/misleading for normal fusion reactors which is what the page is for. Special info about compacts has to go somewhere else.

**Comment by Dream-Master** [comment]
I guess compact and normal fusion use the same recipe map? maybe we need to change that ?

**Comment by HoleFish** [comment]
Rhugnor problem exists in both normals and compacts. You cannot overclock Rhugnor recipe also in normal MK5. It's not sepcial.

**Comment by chochem** [comment]
sounds more like a bug than a documentation gap

**Comment by Dream-Master** [comment]
@chochem so what is about this ?

**Comment by chochem** [comment]
maybe draft for now. we still need some change (code/tooltip/nei/... ?) for compacts.

**Pull request closed without merging**